arpitkh96/AmazeFileManager
# Explicit MIME type declaration on querying available packages for opening files

## Description
As noted by Android Studio on the use of `android.permission.QUERY_ALL_PACKAGES` permission, use `<queries>` with explicitly defined action and MIME type instead. This looks safer (minimal permission principle), and also reduces possibility of unrelated apps from popping up in the list of available apps.

Additionally:
- as noted in https://stackoverflow.com/a/55071463, change SQLITE MIME type to `application/vnd.sqlite3`
- added explicit `android:exported` attribute to `<activity>`s in AndroidManifest.xml where the activity is not necessary to export
- added back `rar` to `FileUtils.COMPRESSED_FILE_EXTENSIONS` to fix problem that opening a RAR file using Amaze does not load up the `CompressedExplorerFragment`

#### Manual tests
  
Device: Pixel 2 emulator
OS: Android 11
Device: Fairphone 3
OS: LineageOS 16.0 (9.0)
Device: Pixel 2 emulator
OS: Android 6.0

- Tapping files (expect those MIME type not explicitly declared in the manifest) should never be affected - user should be given a list of available apps on device for the specified file
- Tapping open a RAR file from Amaze should open `CompressedExplorerFragment` for viewing contents inside the archive

#### Build tasks success  
Successfully running following tasks on local:
